src/libpmemobj/heap.c

```diff
@@ -1302,6 +1357,7 @@ heap_boot(struct palloc_heap *heap, void *heap_start, uint64_t heap_size,
 
 	util_mutex_init(&h->arenas_lock);
 	VEC_INIT(&h->arenas);
+	VEC_RESERVE(&h->arenas, MAX_DEFAULT_ARENAS);
```

Comment:
this can fail. You need to add a new label, and add VEC_DESTROY in between the new one and `error_arenas_malloc`.